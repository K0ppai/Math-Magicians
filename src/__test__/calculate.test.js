import calculate from '../logic/calculate';

const object = {
  total: null,
  next: null,
  operation: null,
};

describe('testing calculate function', () => {
  it('when AC button is clicked, the result should be cleared', () => {
    const buttonName = 'AC';
    const obj = { ...object, next: '5' };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('when next and total is 0, then it should return an empty object', () => {
    const buttonName = '0';
    const obj = { ...object, next: '0', total: '0' };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('when there is no operation, update next and clear the value', () => {
    const buttonName = '5';
    const obj = { ...object };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('when the button is . ', () => {
    const buttonName = '.';
    const obj = { ...object, next: '0.985' };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('when the button is = ', () => {
    const buttonName = '=';
    const obj = { total: '5050', next: '0505', operation: '+' };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('when user puts an operation after pressing = ', () => {
    const buttonName = '-';
    const obj = { ...object, total: '55' };
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });
});
