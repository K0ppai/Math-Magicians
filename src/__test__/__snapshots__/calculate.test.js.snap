// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing calculate function when AC button is clicked, the result should be cleared 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`testing calculate function when next and total is 0, then it should return an empty object 1`] = `Object {}`;

exports[`testing calculate function when the button is .  1`] = `
Object {
  "next": "0.985",
  "operation": null,
  "total": null,
}
`;

exports[`testing calculate function when the button is =  1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "5555",
}
`;

exports[`testing calculate function when there is no operation, update next and clear the value 1`] = `
Object {
  "next": "5",
  "total": null,
}
`;

exports[`testing calculate function when user puts an operation after pressing =  1`] = `
Object {
  "next": null,
  "operation": "-",
  "total": "55",
}
`;
